sgx/core-types/src/report_body.rs

```diff
@@ -288,54 +235,34 @@ impl Ord for ReportBody {
     /// We sort by Family ID, ProdID, Extended ProdID, SVN, MrSigner, MrEnclave, Attributes,
     /// Misc Select, ConfigId, ConfigSVN, CPU SVN, and ReportData, in that order
     fn cmp(&self, other: &Self) -> Ordering {
-        match (&self.0.isv_family_id[..]).cmp(&other.0.isv_family_id[..]) {
-            Ordering::Equal => match self.0.isv_prod_id.cmp(&other.0.isv_prod_id) {
-                Ordering::Equal => match self.0.isv_ext_prod_id.cmp(&other.0.isv_ext_prod_id) {
-                    Ordering::Equal => match self.0.isv_svn.cmp(&other.0.isv_svn) {
-                        Ordering::Equal => match self.mr_signer().cmp(&other.mr_signer()) {
-                            Ordering::Equal => match self.mr_enclave().cmp(&other.mr_enclave()) {
-                                Ordering::Equal => match self.attributes().cmp(&other.attributes())
-                                {
-                                    Ordering::Equal => {
-                                        match self.0.misc_select.cmp(&other.0.misc_select) {
-                                            Ordering::Equal => {
-                                                match self.config_id().cmp(&other.config_id()) {
-                                                    Ordering::Equal => match self
-                                                        .0
-                                                        .config_svn
-                                                        .cmp(&other.0.config_svn)
-                                                    {
-                                                        Ordering::Equal => match self
-                                                            .cpu_security_version()
-                                                            .cmp(&other.cpu_security_version())
-                                                        {
-                                                            Ordering::Equal => self
-                                                                .report_data()
+        (&self.0.isv_family_id[..])
```

Comment:
This looks fine, although it might be more readable to map `self` and `other` into tuples containing the fields you wish to sort by, then use the fact that tuples are ordered lexicographically to do self_as_tuple.cmp(&other_as_tuple). That would avoid a lot of nested comparisons. Maybe use a local function i.e. `as_tuple`, so this becomes something like 

```
fn cmp(...) -> Ordering {
    fn as_tuple(report_body: &ReportBody) -> (x, y, z, ...) {...}
    as_tuple(self).cmp(&as_tuple(other))
}
```